doc/extensions.md

```diff
@@ -0,0 +1,24 @@
+# Working with x.509 Extensions in tk-auth
```

Comment:
Where the `authorization-config.md` doc is starting to get pretty long, having a separate document that just talks about the extensions - especially as we may intend to add more standard ones in the future - seems like a good idea.  I do think it would be good to link from `authorization-config.md` back to this doc if we intend to do that, though.  Maybe we could embed links from here - https://github.com/puppetlabs/trapperkeeper-authorization/blob/0.6.0/doc/authorization-config.md#allow - and here - https://github.com/puppetlabs/trapperkeeper-authorization/blob/0.6.0/doc/authorization-config.md#deny?